pyiron_nodes/atomistic/engine/vasp.py

```diff
@@ -0,0 +1,263 @@
+                    return True
+                elif not exact_match and line in file_line:
+                    return True
+    except FileNotFoundError:
+        print(f"File '{filepath}' not found.")
+    return False
+
+
+def write_POSCAR(workdir: str, structure: Structure, filename: str = "POSCAR") -> str:
+    poscar_path = os.path.join(workdir, filename)
+    structure.to(fmt="poscar", filename=poscar_path)
+    return poscar_path
+
+
+def write_INCAR(workdir: str, incar: Incar, filename: str = "INCAR") -> str:
+    incar_path = os.path.join(workdir, filename)
+    incar.write_file(incar_path)
+    return incar_path
+
+
+def write_POTCAR(workdir: str, vasp_input, filename: str = "POTCAR") -> str:
+    class PotcarNotGeneratedError(Exception):
+        pass
+
+    if vasp_input.potcar_paths is None:
+        potcar_paths = get_default_POTCAR_paths(
+            vasp_input.structure, vasp_input.pseudopot_lib_path
+        )
+    else:
+        potcar_paths = vasp_input.potcar_paths
+
+    potcar_path = os.path.join(workdir, filename)
+
+    with open(potcar_path, "wb") as wfd:
+        for f in potcar_paths:
+            with open(f, "rb") as fd:
+                shutil.copyfileobj(fd, wfd)
+
+    return potcar_path
+
+
+def write_KPOINTS(
+    workdir: str, kpoints: Optional[Kpoints] = None, filename: str = "KPOINTS"
+) -> str:
+    kpoint_path = os.path.join(workdir, filename)
+    if kpoints is not None:
+        kpoints.write_file(kpoint_path)
+    else:
+        with open(kpoint_path, "w") as f:
+            f.write(
+                "Automatic mesh\n0\nGamma\n1 1 1\n0 0 0\n"
+            )  # Example KPOINTS content
+    return kpoint_path
+
+
+@Workflow.wrap.as_function_node("workdir")
+def create_WorkingDirectory(workdir: str, quiet: bool = False) -> str:
+    # Check if workdir exists
+    if not os.path.exists(workdir):
+        os.makedirs(workdir)
+        print(f"made directory '{workdir}'")
+    else:
+        warnings.warn(
+            f"Directory '{workdir}' already exists. Existing files may be overwritten."
+        )
+    return workdir
+
+
+@Workflow.wrap.as_function_node("workdir")
+def write_VaspInputSet(workdir: str, vasp_input) -> str:
+    _ = write_POSCAR(workdir=workdir, structure=vasp_input.structure)
+    _ = write_INCAR(workdir=workdir, incar=vasp_input.incar)
+    _ = write_POTCAR(workdir=workdir, vasp_input=vasp_input)
+    if vasp_input.kpoints is not None:
+        _ = write_KPOINTS(workdir=workdir, kpoints=vasp_input.kpoints)
+
+    return workdir
+
+
+class Storage:
+    @staticmethod
+    def _convert_to_dict(instance) -> Dict:
+        attributes = vars(instance)
+        result_dict = {
+            key: value for key, value in attributes.items() if "_" not in key[0]
+        }
+        return result_dict
+
+
+class ShellOutput(Storage):
+    stdout: str
+    stderr: str
+    return_code: int
+
+
+@Workflow.wrap.as_function_node("output")
+def run_job(
```

Comment:
This is super nice. I wonder how it interacts with `mpi -n` commands, and whether we could add some extra inputs to the node and then automatically modify the command ala [my `_maybe_mpi` function](https://github.com/liamhuber/scipy2024/blob/6c047f5b5e3a7f7f803b3c6cdb7768891582b7c5/ase_nodes/calculators.py#L57).

Otherwise my only complaint is that this is super generic and shouldn't be hidden away in `pyiron_nodes.atomistic.engine.vasp`. This belongs in `pyiron_nodes.shell` or something similarly high-level! I also know Joerg has done some stuff in this direction, but am not familiar enough with it to know by-heart how this compares. We should go back and look for overlap, maybe merging the best features between the two if he indeed has a similar node.